WooCommerce/Classes/ViewRelated/Orders/OrdersSplitViewWrapperController.swift

```diff
@@ -97,10 +97,33 @@ private extension OrdersSplitViewWrapperController {
             return
         }
 
-        let orderDetailsViewController = OrderDetailsViewController(viewModels: viewModels, currentIndex: currentIndex)
-        let orderDetailsNavigationController = WooNavigationController(rootViewController: orderDetailsViewController)
+        let orderDetailsViewController = OrderDetailsViewController(
+            viewModels: viewModels,
+            currentIndex: currentIndex,
+            switchDetailsHandler: { [weak self] viewModels, currentIndex, isSelectedManually, completion in
+                self?.handleSwitchingDetails(viewModels: viewModels,
+                                             currentIndex: currentIndex,
+                                             isSelectedManually: isSelectedManually,
+                                             onCompletion: completion)
+            })
+
+        // When showing an order with multiple order view models to support quick navigation (up and down arrows),
+        // subsequent order details should replace the top order details view controller to avoid multiple order details
+        // in the navigation stack.
+        guard viewModels.count > 1,
```

Comment:
We could extract this to a bool named `hasQuickOrderNavigation` or similar, to add some meaning in the code.

Reply:
Good point! updated in https://github.com/woocommerce/woocommerce-ios/pull/11813/commits/4f51e35188f6da83c1b86f47d110beb2164bbd63